### PR TITLE
Cold gas plots wishlist

### PR DESCRIPTION
**Pull request by wmfw23**
most recent plots requested not included

### CODE REVIEW

File: colibre/auto_plotter/cold_gas_relations.yml [hunk 1]
```diff
@@ -22,41 +22,18 @@ stellar_mass_molecular_to_neutral_fraction_30:
     end:
       value: 1e12
       units: Solar_Mass
+    lower:
+      value: 0
+      units: "dimensionless"
+    upper:
+      value: 1e99
+      units: "dimensionless"
```
Comment (JBorrow) [inline colibre/auto_plotter/cold_gas_relations.yml:30]: I'm still not very happy with this... I would prefer it if we're not including galaxies in the median line then there should be a 'physical' reasoning for it, e.g. we only want to include SFing galaxies

File: colibre/auto_plotter/cold_gas_relations.yml [hunk 2]
```diff
@@ -350,4 +354,534 @@ H2_mass_star_formation_rate_100:
...
+  y:
+    quantity: "derived_quantities.gas_HI_to_neutral_H_fraction_100_kpc"
+    units: "dimensionless"
+    start: 1e-1
+    end: 1.1
+  median:
+    plot: true
+    log: true
+    adaptive: true
+    number_of_bins: 15
```
Comment (JBorrow) [inline colibre/auto_plotter/cold_gas_relations.yml:411]: We probably want to keep the bins consistent with what we do in e.g. the mass function here, at least by default.

File: colibre/auto_plotter/cold_gas_relations.yml [hunk 3]
```diff
@@ -350,4 +354,534 @@ H2_mass_star_formation_rate_100:
...
+    lower:
+      value: 0
+      units: "dimensionless"
+    upper:
+      value: 1e99
+      units: "dimensionless"
+  metadata:
+    title: Stellar Mass-H2 to Neutral Gas Fraction (100 kpc Stellar Mass)
+    caption: All galaxies are included in the median line. Fraction is H2 mass over HI + H2 mass in a 100 kpc aperture.
+    section: Cold Gas Fractions vs M*
```
Comment (JBorrow) [inline colibre/auto_plotter/cold_gas_relations.yml:462]: Text should be well-formatted everywhere; can we do `Cold Gas Fractions (Stellar Mass)` or `Cold Gas Fractions ($M_*$)`?

File: colibre/auto_plotter/cold_gas_relations.yml [hunk 4]
```diff
@@ -261,7 +263,7 @@ stellar_mass_HI_mass_30:
   metadata:
     title: Stellar Mass-HI Gas Mass (30 kpc Stellar Mass)
     caption: All galaxies are included in the median line.
-    section: Cold Gas Relations
+    section: Original Cold Gas Relations
```
Comment (JBorrow) [inline colibre/auto_plotter/cold_gas_relations.yml:266]: The name original doesn't really make very much sense here. I guess all of these are against gas mass? Can we do `Cold Gas Relations (Gas)` or `Cold Gas Relations ($M_{\rm g}$)`?

File: colibre/auto_plotter/cold_gas_relations.yml [hunk 5]
```diff
@@ -350,4 +354,534 @@ H2_mass_star_formation_rate_100:
...
+      value: 10
+      units: "1 / Gyr"
+    lower:
+      value: 0
+      units: "dimensionless"
+    upper:
+      value: 1e99
+      units: "dimensionless"
+  metadata:
+    title: sSFR-H2 Gas to SF Gas Fraction (100 kpc Stellar Mass)
```
Comment (JBorrow) [inline colibre/auto_plotter/cold_gas_relations.yml:708]: Please format the titles using LaTeX where appropriate, e.g. `sSFR-H$_2$` here.

File: colibre/auto_plotter/cold_gas_relations.yml [hunk 6]
```diff
@@ -350,4 +354,534 @@ H2_mass_star_formation_rate_100:
...
+    caption: All galaxies are included in the median line. Fraction is H2 mass over stellar mass in a 100 kpc aperture.
+    section: Cold Gas Fractions vs sSFR
```
Comment (JBorrow) [inline colibre/auto_plotter/cold_gas_relations.yml:710]: Keep this consistent with the rest `Cold Gas Fractions (sSFR)`

Comment (JBorrow) [reply colibre/auto_plotter/cold_gas_relations.yml:710]: I don't follow in this section why we need the vertical bounding on the median line, too.

File: colibre/auto_plotter/cold_gas_relations.yml [hunk 7]
```diff
@@ -350,4 +354,534 @@ H2_mass_star_formation_rate_100:
...
+    lower:
+      value: 0
+      units: "dimensionless"
+    upper:
+      value: 1e99
+      units: "dimensionless"
+  metadata:
+    title: Surface Density-SF Gas to SF Gas + Stellar Fraction (100 kpc Stellar Mass)
+    caption: All galaxies are included in the median line. Fraction is SF gas mass over SF gas mass + Stellar mass in a 100 kpc aperture.
+    section: Cold Gas Fractions vs Surface Density
```
Comment (JBorrow) [inline colibre/auto_plotter/cold_gas_relations.yml:887]: Again, section name should be consistent with the rest `Cold Gas Fractions ($\mu_*$)`

File: colibre/auto_plotter/gas_fraction_plots.yml [hunk 8]
```diff
@@ -72,10 +72,10 @@ h2_frac_func_stellar_surface_density:
   x:
     quantity: "derived_quantities.mu_star_100_kpc"
     units: "Solar_Mass / (kpc**2)"
-    start: 1e9
+    start: 1e7
```
Comment (JBorrow) [inline colibre/auto_plotter/gas_fraction_plots.yml:75]: You should change the number of bins here too, to make sure it's all consistent with the standard 0.2 dex.

File: colibre/auto_plotter/gas_mass_functions.yml [hunk 9]
```diff
@@ -47,7 +47,7 @@ gas_H2_mass_function:
   legend_loc: "lower left"
   number_of_bins: 25
   x:
-    quantity: "derived_quantities.gas_H2_mass_Msun"
+    quantity: "derived_quantities.gas_H2_plus_He_mass_Msun"
```
Comment (JBorrow) [inline colibre/auto_plotter/gas_mass_functions.yml:50]: Why the `Msun`? The unit information is carried in the `unyt` metadata.

File: colibre/auto_plotter/gas_fraction_plots.yml (hunk 8 above)
Comment (wmfw23) [reply colibre/auto_plotter/gas_fraction_plots.yml:75]: I've set all to 0.2 dex by default - but i think this is probably too narrow for runs particularly in small boxes as any high dispersion plots can yield very noisy trendlines

Comment (JBorrow) [reply colibre/auto_plotter/gas_fraction_plots.yml:75]: That's why we use the adaptive lines, so hopefully we shouldn't have a problem.

**Comment by JBorrow** [comment]
Sorry, I think you're going to need to rebase the registration file :(

**Comment by MatthieuSchaller** [comment]
I can do the porting over to the XL side when accepted.

**Comment by wmfw23** [comment]
@JBorrow I can do that, though I think I have to rebase the whole branch, right?

**Comment by JBorrow** [comment]
Yeah, or you can use the `Resolve Conflicts` button on GitHub and do it that way.

**Comment by wmfw23** [comment]
rebase seemed to work, but now there's a problem with the observational data - autoplotter is looking for `GalaxyStellarMassHaloMass/Schaye2015_Ref_100_M200.hdf5` but the observational data files have redshift suffixes here. Does the submodule need updating in the pipeline branch or something? I have the commit `a2bc3342e8109898d2ac47b1e417e24e1aa561a4` which says sometihng about stripping suffixes.

**Comment by MatthieuSchaller** [comment]
You may need to update the submodule to the latest version used in master and then regenerate the observational data.

**Comment by wmfw23** [comment]
Yeah the submodules always get me, i've pushed that now

**Comment by MatthieuSchaller** [comment]
Great!

**Comment by MatthieuSchaller** [comment]
At the risk of being extra pedantic, would it make sense to mention the He correction in the relevant plot captions?

**Comment by wmfw23** [comment]
@MatthieuSchaller yes sure, it's mentioned in the axis label but should probably be in the caption too

**Comment by MatthieuSchaller** [comment]
Looks good to me with one caveat. Wouldn't
```
HI fraction over stellar mass as a function ...
```

read better as
```
HI mass over stellar mass as a function ...
```

i.e. the first one could be understood as being `(M_HI / X ) / M_* ` as a function of YYY.

**Comment by MatthieuSchaller** [comment]
@JBorrow anything stopping this to be merged?

File: colibre/registration.py [hunk 10]
```diff
@@ -238,6 +244,48 @@
...
+        HI_to_sf_fraction.name = (
+            f"$M_{{\\rm HI}}/M_{{\\rm SF}}$ ({aperture_size} kpc)"
+        )
+
+        H2_to_sf_fraction = H2_mass / sf_mass
+        H2_to_sf_fraction.name = (
+            f"$M_{{\\rm H2}}/M_{{\\rm SF}}$ ({aperture_size} kpc)"
+        )
+
+        sf_to_stellar_fraction = H2_mass / sf_mass
```
Comment (EvgeniiChaikin) [inline colibre/registration.py:284]: Hi @wmfw23 I have noticed that here the name seems to be inconsistent with what is plotted

**Pull request merged**